benches/benches/bevy_ecs/change_detection.rs

```diff
@@ -243,3 +265,109 @@ fn none_changed_detection(criterion: &mut Criterion) {
+            insert_if_bit_enabled::<5>(&mut e, i);
+            insert_if_bit_enabled::<6>(&mut e, i);
+            insert_if_bit_enabled::<7>(&mut e, i);
+            insert_if_bit_enabled::<8>(&mut e, i);
+            insert_if_bit_enabled::<9>(&mut e, i);
+            insert_if_bit_enabled::<10>(&mut e, i);
+            insert_if_bit_enabled::<11>(&mut e, i);
+            insert_if_bit_enabled::<12>(&mut e, i);
+            insert_if_bit_enabled::<13>(&mut e, i);
+            insert_if_bit_enabled::<14>(&mut e, i);
+            insert_if_bit_enabled::<15>(&mut e, i);
+        }
+    }
+}
+fn multiple_archetype_none_changed_detection_generic<T: Component + Default + BenchModify>(
+    group: &mut BenchGroup,
+    archetype_count: u16,
+    entity_count: u32,
+) {
+    group.bench_function(
+        format!(
+            "{}_archetypes_{}_entities_{}",
+            archetype_count,
+            entity_count,
+            std::any::type_name::<T>()
+        ),
+        |bencher| {
+            bencher.iter_batched_ref(
+                || {
+                    let mut world = World::new();
+                    add_archetypes_entites::<T>(&mut world, archetype_count, entity_count);
+                    world.clear_trackers();
+                    let mut query = world.query::<(
+                        Option<&mut Data<0>>,
+                        Option<&mut Data<1>>,
+                        Option<&mut Data<2>>,
+                        Option<&mut Data<3>>,
+                        Option<&mut Data<4>>,
+                        Option<&mut Data<5>>,
+                        Option<&mut Data<6>>,
+                        Option<&mut Data<7>>,
+                        Option<&mut Data<8>>,
+                        Option<&mut Data<9>>,
+                        Option<&mut Data<10>>,
+                        Option<&mut Data<11>>,
+                        Option<&mut Data<12>>,
+                        Option<&mut Data<13>>,
+                        Option<&mut Data<14>>,
+                    )>();
+                    for components in query.iter_mut(&mut world) {
+                        modify!(components;0,1,2,3,4,5,6,7,8,9,10,11,12,13,14);
+                    }
+                    let query = generic_filter_query::<Changed<T>>(&mut world);
+                    (world, query)
+                },
+                |(ref mut world, ref mut query)| {
+                    let mut count = 0;
+                    for entity in query.iter(&world) {
+                        black_box(entity);
+                        count += 1;
+                    }
+                    assert_eq!(0, count);
+                },
+                criterion::BatchSize::LargeInput,
+            )
+        },
+    );
+}
+
+fn multiple_archetype_none_changed_detection(criterion: &mut Criterion) {
+    let mut group = criterion.benchmark_group("multiple_archetypes_none_changed_detection");
+    group.warm_up_time(std::time::Duration::from_millis(800));
```

Comment:
How come we changed the warm_up_time and measurement_time for this benchmark?

Reply:
emm,IMO, It may require longer time to obtain more data when the iteration needs to spawn 100 * 10000 entities.